extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -0,0 +1,378 @@
+   */
+  update(unusedElement) {
+  }
+
+  /**
+   * The interval of when the LiveList should be at best notified
+   * of new updates.
+   *
+   * @return {number}
+   */
+  getInterval() {}
+}
+
+
+/**
+ * Helper function that either returns a number derived from the given
+ * string value or returns the default value passed in.
+ *
+ * @param {string} value
+ * @param {number} defaultValue
+ * @return {number}
+ */
+function getNumberOrDefault(value, defaultValue) {
+  value = Number(value);
+  return value > 0 ? Math.max(value, defaultValue) : defaultValue;
+}
+
+
+/**
+ * Component class that handles updates to its underlying children dom
+ * structure.
+ * @implements {LiveListInterface}
+ */
+export class AmpLiveList extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @const {!Window} */
+    this.win = this.getWin();
+
+    /** @private @const {boolean} */
+    this.isExperimentOn_ = isExperimentOn(this.win, TAG);
+
+    if (!this.isExperimentOn_) {
+      user.warn(TAG, `Experiment ${TAG} disabled`);
+      return;
+    }
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportFor(this.win);
+
+    /** @private @const {!LiveListManager} */
+    this.manager_ = installLiveListManager(this.win);
+
+    /** @private @const {string} */
+    this.liveListId_ = user.assert(this.element.getAttribute('id'),
+        'amp-live-list must have an id.');
+
+    /** @private @const {number} */
+    this.pollInterval_ = getNumberOrDefault(
+        this.element.getAttribute('data-poll-interval'),
+        LiveListManager.getMinDataPollInterval());
+
+    const maxItems = this.element.getAttribute('data-max-items-per-page');
+    user.assert(Number(maxItems) > 0,
+        `amp-live-list#${this.liveListId_} must have ` +
+        `data-max-items-per-page attribute with numeric value. ` +
+        `Found ${maxItems}`);
+
+    /** @private @const {number} */
+    this.maxItemsPerPage_ = getNumberOrDefault(maxItems,
+        LiveListManager.getMinDataMaxItemsPerPage());
+
+    /** @private @const {!Object<string, string>} */
+    this.knownChildIds_ = Object.create(null);
+
+    this.manager_.register(this.liveListId_, this);
+
+    this.insertFragment_ = this.win.document.createDocumentFragment();
+
+    this.updateSlot_ = user.assert(
+       this.getUpdateSlot_(this.element),
+       'amp-live-list must have an `update` slot.');
+
+    this.itemsSlot_ = user.assert(
+        this.getItemsSlot_(this.element),
+        'amp-live-list must have an `items` slot.');
+
+    this.updateSlot_.classList.add('-amp-hidden');
+    this.eachChildElement_(this.itemsSlot_, item => {
+      item.classList.add(classes.ITEM);
+    });
+
+    this.validateLiveListItems_(this.itemsSlot_, true);
+
+    this.registerAction('update', this.updateAction_.bind(this));
+  }
+
+  /** @override */
+  update(updatedElement) {
```

Comment:
it is not. not sure why we need `vsync.mutate` here since we only manipulate the documentFragment and not the live dom.